modules/delegation-renewal/hooks/useDelegatedToExpired.tsx

```diff
@@ -0,0 +1,23 @@
+import { add, isBefore } from 'date-fns';
+import useDelegatedTo from 'modules/delegates/hooks/useDelegatedTo';
+import { useActiveWeb3React } from 'modules/web3/hooks/useActiveWeb3React';
+
+export default function useDelegatedToExpired(): boolean {
+  const { account: address, network } = useActiveWeb3React();
+
+  const { data: delegatedToData } = useDelegatedTo(address, network);
+
+  const isADelegateAboutToExpire = delegatedToData
+    ? delegatedToData.delegatedTo.reduce((prev, next) => {
+        // check if it has less than one week to expire or is expired
```

Comment:
We need to distinguish between `expired` or `expiring soon` so we can use that logic throughout the app for the banner and different flows.